javascript/tables/src/validateTable.ts

```diff
@@ -0,0 +1,288 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type {
+	CodebookSchema,
+	ColumnError,
+	Field,
+	ValidationResult,
+} from '@datashaper/schema'
+import { DataType } from '@datashaper/schema'
+import type ColumnTable from 'arquero/dist/types/table/column-table'
+
+export function validateTable(
+	table: ColumnTable,
+	codebook: CodebookSchema,
+): ValidationResult {
+	const validationResult: ValidationResult = {
+		columnErrors: [],
+	}
+
+	codebook.fields.forEach((field: Field) => {
+		const columnValues = table.array(field.name)
```

Comment:
You might want to confirm if a field has any constraints defined _before_ calling `table.array` to avoid lots of unnecessary table scans. I would just move the `if` statement below up outside of this.